libs/application-generic/src/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts

```diff
@@ -49,30 +60,138 @@ export class GetSubscriberPreference {
+          type: {
+            $in: [
+              PreferencesTypeEnum.WORKFLOW_RESOURCE,
+              PreferencesTypeEnum.USER_WORKFLOW,
+            ],
+          },
+        },
+        {
+          _subscriberId: subscriber._id,
+          type: PreferencesTypeEnum.SUBSCRIBER_GLOBAL,
+        },
+      ],
+    });
+
+    const subscriberGlobalPreferences = allPreferences.filter(
+      (preference) =>
+        preference._subscriberId === subscriber._id &&
+        preference.type === PreferencesTypeEnum.SUBSCRIBER_GLOBAL,
+    );
+
+    const workflowPreferenceSets = allPreferences.reduce<
+      Record<string, PreferencesEntity[]>
+    >((acc, preference) => {
+      const workflowId = preference._templateId;
+
+      // Skip if the preference is not for a workflow
+      if (workflowId === undefined) {
+        return acc;
+      }
+
+      if (!acc[workflowId]) {
+        acc[workflowId] = [];
+      }
+      acc[workflowId].push(preference);
+
+      return acc;
+    }, {});
+
+    const mergedPreferences: ISubscriberPreferenceResponse[] = workflowList.map(
+      (workflow) => {
+        const preferences = workflowPreferenceSets[workflow._id] || [];
+        const mergeCommand = MergePreferencesCommand.create({
+          preferences: [...preferences, ...subscriberGlobalPreferences],
+        });
+        const merged = MergePreferences.merge(mergeCommand);
+
+        const includedChannels = this.getChannels(
+          workflow,
+          command.includeInactiveChannels,
+        );
+
+        const initialChannels = filteredPreference(
+          {
+            email: true,
+            sms: true,
+            in_app: true,
+            chat: true,
+            push: true,
+          },
+          includedChannels,
+        );
+
+        const { channels, overrides } = overridePreferences(
+          {
+            template: GetPreferences.mapWorkflowPreferencesToChannelPreferences(
+              merged.source.WORKFLOW_RESOURCE,
+            ),
+            subscriber:
+              GetPreferences.mapWorkflowPreferencesToChannelPreferences(
+                merged.preferences,
+              ),
+            workflowOverride: {},
+          },
+          initialChannels,
+        );
+
+        return {
+          preference: {
+            channels,
+            enabled: true,
+            overrides,
+          },
+          template: {
+            ...workflow,
+            critical: merged.preferences.all.readOnly,
+          },
+          type: PreferencesTypeEnum.SUBSCRIBER_WORKFLOW,
+        };
+      },
     );
 
-    const nonCriticalWorkflowPreferences = subscriberWorkflowPreferences.filter(
-      (preference) => preference.template.critical === false,
+    const nonCriticalWorkflowPreferences = mergedPreferences.filter(
+      (preference) => !preference.template.critical,
     );
 
     return nonCriticalWorkflowPreferences;
   }
+
+  private getChannels(
```

Comment:
I have left it as `getChannels` because it's possible for the method to return inactive channels when specifying the `includeInactiveChannels` flag, and the refined naming would be misleading.